docs/introduction/concepts.md

```diff
@@ -0,0 +1,281 @@
+        return TaskCreated
+                .newBuilder()
+                .setId(cmd.getId())
+                .setName(cmd.getName())
+                .setOwner(ctx.getActor())
+                .vBuild();
+    }
+    ...
+}
+```
+
+### Event Subscriber
+
+Event Subscriber is an object which subscribes to receive events.
+
+The example below shows how a [Projection](#projection) class subscribed to the `TaskCompleted`
+event.
+
+  ```
+  final class TaskProjection
+      extends Projection<TaskId, TaskItem, TaskItem.Builder> {
+      ...
+      @Subscribe
+      void on(TaskCompleted e, EventContext ctx) {
+          builder().setWhenDone(ctx.getTimestamp());
+      }
+  }
+  ```
+
+### Event Reactor
+
+Event Reactor is an object which usually produces one or more events in response to an incoming
+event. Unlike [Event Subscriber](#event-subscriber), which always consumes events, a reacting object
+generates events in response to changes in the domain.
+
+<p class="note">In some cases, Event Reactor may ignore the event, returning `Nothing`.
+    It usually happens when a method returns one of the
+    [`Either`](https://spine.io/core-java/javadoc/server/index.html) types, with `Nothing` as
+    one of the possible options: `EitherOf2<TaskReAssigned, Nothing>`.</p>
+
+## Value Objects
+
+Value Object describe things in a domain model and do not have identity. 
+Value Objects are also immutable. Some examples are:
+  * `PhoneNumber`
+  * `EmailAddress`
+  * `BarCode`
+  
+In Spine, Value Objects are defined as Protobuf messages.
+
+## Entities
+
+Entities are main building blocks of a domain model. They have unique identity and modify their 
+state during the lifecycle.
+
+### Identifier
+ 
+The framework supports the following types of identifiers:
+
+* `Integer`,
+* `Long`,
+* `String`,
+* A generated Java class implementing the `Message`.
+
+Examples of entity IDs used by the framework: `CommandId`, `EventId`, `UserId`, `TenantId`.
+
+<p class="note">We highly recommend using message-based IDs to make your API strongly pronounced
+    and type-safe.</p>
+
+### Aggregate
+
+Aggregate is the main building block of a business model. 
+From the application point of view it consists of the following:
+1. Commands which arrive to it. 
+2. Events which appear in response to these commands. 
+3. How these events influence the state of an aggregate.
+
+[Aggregates](http://martinfowler.com/bliki/DDD_Aggregate.html) guarantee consistency of data
+modifications in response to commands they receive. Aggregate is the most common case of
+Command Handler. It modifies its state and produces one or more events in response to a command.
+These events are used later to restore the state of the aggregate.
+
+In Spine, aggregates are defined as Java classes, and their states are defined as Protobuf messages.
+
+### Process Manager
+
+Process Manager is an independent component that reacts to domain events in a cross-aggregate
```

Comment:
I would rephrase.

```
Process Manager is an independent component that manages the cross-aggregate business flows. 
It serves as a mediator by remembering the state of the flow and choosing the next step 
based on the intermediate results.

To do so, a Process Manager can be both [Command Handler](#command-handler) 
and [Event Reactor](#event-reactor). Also, it can emit Commands to other Aggregates 
and Process Managers.
```